File: app/src/main/java/com/yinp/proappkotlin/base/adapter/ComViewHolder.java
package com.yinp.proappkotlin.base.adapter;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

/**
 * 通用ViewHolder
 */
public abstract class ComViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {
    public ComViewHolder(View itemView) {
        super(itemView);
        itemView.setOnClickListener(this);
        itemView.setOnLongClickListener(this);
        init(itemView);
    }

    public void init(View view) {

    }

    OnItemClickListener onItemClickListener;

    @Override
    public void onClick(View v) {
        if (onItemClickListener != null) {
            onItemClickListener.onItemClick(getLayoutPosition(), v);
        }
    }

    public void onItemClickListener(OnItemClickListener listener) {
        this.onItemClickListener = listener;
    }

    @Override
    public boolean onLongClick(View v) {
        return false;
    }

    public interface OnItemClickListener {
        void onItemClick(int position, View view);
    }
}

File: tools/src/main/java/com/yinp/tools/adapter/ComViewHolder.kt
package com.yinp.tools.adapter

import android.view.View
import android.view.View.OnLongClickListener
import androidx.recyclerview.widget.RecyclerView

/**
 * @author   :yinpeng
 * date      :2021/10/14
 * package   :com.yinp.tools.adapter
 * describe  :
 */
abstract class ComViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),
    View.OnClickListener, OnLongClickListener {
    init {
        itemView.setOnClickListener(this)
        itemView.setOnLongClickListener(this)
    }

    private var mOnItemClickListener: OnItemClickListener? = null
    override fun onClick(v: View?) {
        if (mOnItemClickListener != null) {
            mOnItemClickListener!!.onItemClick(layoutPosition, v)
        }
    }

    fun onItemClickListener(listener: OnItemClickListener?) {
        mOnItemClickListener = listener
    }

    override fun onLongClick(v: View?): Boolean {
        return false
    }

    interface OnItemClickListener {
        fun onItemClick(position: Int, view: View?)
    }


}